insights/2015-04-01/swagger/insightsClient_TenantEvents.json

```diff
@@ -0,0 +1,297 @@
+{
+  "swagger": "2.0",
+  "info": {
+    "title": "InsightsClient",
+    "version": "2015-04-01"
+  },
+  "host": "management.azure.com",
+  "schemes": [
+    "https"
+  ],
+  "consumes": [
+    "application/json"
+  ],
+  "produces": [
+    "application/json"
+  ],
+  "paths": {
+    "/providers/microsoft.insights/eventtypes/management/values": {
+      "get": {
+        "tags": [
+          "TenantEvents"
+        ],
+        "operationId": "TenantEvents_List",
+        "description": "get the Activity Logs for the Tenant. Everything that is applicable to the API to get the Activity Log for the subscription is applicable to this API (the parameters, $filter, etc.). One thing to point out here is that this API does *not* retrieve the logs at the individual subscription of the tenant but only surfaces the logs that were generated at the tenant level. The **$filter** is very restricted and allows only the following patterns. - List events for a resource group: $filter=eventTimestamp ge '<Start Time>' and eventTimestamp le '<End Time>' and eventChannels eq 'Admin, Operation' and resourceGroupName eq '<ResourceGroupName>'. - List events for resource: $filter=eventTimestamp ge '<Start Time>' and eventTimestamp le '<End Time>' and eventChannels eq 'Admin, Operation' and resourceUri eq '<ResourceURI>'. - List events for a subscription: $filter=eventTimestamp ge '<Start Time>' and eventTimestamp le '<End Time>' and eventChannels eq 'Admin, Operation'. - List evetns for a resource provider: $filter=eventTimestamp ge '<Start Time>' and eventTimestamp le '<End Time>' and eventChannels eq 'Admin, Operation' and resourceProvider eq '<ResourceProviderName>'. - List events for a correlation Id: api-version=2014-04-01&$filter=eventTimestamp ge '2014-07-16T04:36:37.6407898Z' and eventTimestamp le '2014-07-20T04:36:37.6407898Z' and eventChannels eq 'Admin, Operation' and correlationId eq '<CorrelationID>'. No other syntax is allowed.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/FilterParameter"
+          },
+          {
+            "$ref": "#/parameters/SelectParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful request to get a page of tenant events",
+            "schema": {
+              "$ref": "#/definitions/EventDataCollection"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
```

Comment:
It is ok to not specify null and specify `"nextLinkName": "nextLink"`. This will pave the way for the future. Whenever the service decides to make this a pageable operation it wont be a breaking change in the generated code.